packages/gatsby/src/commands/develop.ts

```diff
@@ -1,631 +1,240 @@
-import cors from "cors"
-import telemetry from "gatsby-telemetry"
-import * as WorkerPool from "../utils/worker/pool"
+// NOTE(@mxstbr): Do not use the reporter in this file, as that has side-effects on import which break structured logging
+import path from "path"
 import http from "http"
-import https from "https"
-
-import {
-  bootstrapSchemaHotReloader,
-  startSchemaHotReloader,
-  stopSchemaHotReloader,
-} from "../bootstrap/schema-hot-reloader"
-import bootstrapPageHotReloader from "../bootstrap/page-hot-reloader"
-import { developStatic } from "./develop-static"
-import withResolverContext from "../schema/context"
-import sourceNodes from "../utils/source-nodes"
-import { createSchemaCustomization } from "../utils/create-schema-customization"
-import { rebuild as rebuildSchema } from "../schema"
-import { websocketManager } from "../utils/websocket-manager"
-import getSslCert from "../utils/get-ssl-cert"
-import { slash } from "gatsby-core-utils"
-import { initTracer } from "../utils/tracer"
-import apiRunnerNode from "../utils/api-runner-node"
-import db from "../db"
-import { detectPortInUseAndPrompt } from "../utils/detect-port-in-use-and-prompt"
-import onExit from "signal-exit"
-import queryUtil from "../query"
-import queryWatcher from "../query/query-watcher"
-import requiresWriter from "../bootstrap/requires-writer"
-import {
-  reportWebpackWarnings,
-  structureWebpackErrors,
-} from "../utils/webpack-error-utils"
-import { waitUntilAllJobsComplete } from "../utils/wait-until-jobs-complete"
-import {
-  userPassesFeedbackRequestHeuristic,
-  showFeedbackRequest,
-} from "../utils/feedback"
-
-import { IPreparedUrls, prepareUrls } from "../utils/prepare-urls"
-import { Stage, IProgram } from "./types"
-
-// checks if a string is a valid ip
-const REGEX_IP = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$/
-
-// const isInteractive = process.stdout.isTTY
-
-// Watch the static directory and copy files to public as they're added or
-// changed. Wait 10 seconds so copying doesn't interfere with the regular
-// bootstrap.
-setTimeout(() => {
-  syncStaticDir()
-}, 10000)
-
-onExit(() => {
-  telemetry.trackCli(`DEVELOP_STOP`)
-})
-
-type ActivityTracker = any // TODO: Replace this with proper type once reporter is typed
-
-interface IServer {
-  compiler: webpack.Compiler
-  listener: http.Server | https.Server
-  webpackActivity: ActivityTracker
-}
+import tmp from "tmp"
+import { spawn } from "child_process"
+import chokidar from "chokidar"
+import getRandomPort from "get-port"
+import socket from "socket.io"
+import fs from "fs-extra"
+import { isCI } from "gatsby-core-utils"
+import { createServiceLock } from "gatsby-core-utils/dist/service-lock"
+import { startDevelopProxy } from "../utils/develop-proxy"
+import { IProgram } from "./types"
+
+// Adapted from https://stackoverflow.com/a/16060619
+const requireUncached = (file: string): any => {
+  try {
```

Comment:
Nit: I believe this can be one try-catch block

Reply:
It actually can't be! Even if we fail to delete from the cache (which happens when the module hasn't been required yet) we want to return the required file afterwards (which might still exist) :blush:

Reply:
oh cool, I thought it was failing because of require.resolve :)